packages/components/src/TimeInput.test.tsx

```diff
@@ -181,20 +164,42 @@ describe('arrow left and right jumps segments', () => {
     movement: number | number[],
     expectedSelection: SelectionSegment
   ) {
-    const timeInput = makeTimeInput();
-
-    selectRange(timeInput, makeSelection(cursorPosition, cursorPosition));
+    const { unmount } = makeTimeInput();
+    const input: HTMLInputElement = screen.getByRole('textbox');
+    input.focus();
+    input.setSelectionRange(cursorPosition, cursorPosition);
+    userEvent.type(input, '', {
+      initialSelectionStart: cursorPosition,
+      initialSelectionEnd: cursorPosition,
+    });
 
     const movements: number[] = ([] as number[]).concat(movement);
-    const inputField = timeInput.find('input');
+
     for (let i = 0; i < movements.length; i += 1) {
       const arrowMovement = movements[i];
+
       for (let j = 0; j < arrowMovement; j += 1) {
-        inputField.simulate('keydown', { key: 'ArrowRight' });
+        fireEvent.keyDown(input, {
+          key: 'ArrowRight',
+        });
+        fireEvent.keyPress(input, {
+          key: 'ArrowRight',
+        });
+        fireEvent.keyUp(input, {
+          key: 'ArrowRight',
+        });
       }
 
       for (let j = 0; j > arrowMovement; j -= 1) {
-        inputField.simulate('keydown', { key: 'ArrowLeft' });
+        fireEvent.keyDown(input, {
+          key: 'ArrowLeft',
+        });
+        fireEvent.keyPress(input, {
+          key: 'ArrowLeft',
+        });
+        fireEvent.keyUp(input, {
+          key: 'ArrowLeft',
+        });
```

Comment:
Should be able to shorten this down with formatting
```suggestion
        fireEvent.keyDown(input, { key: 'ArrowLeft' });
        fireEvent.keyPress(input, { key: 'ArrowLeft' });
        fireEvent.keyUp(input, { key: 'ArrowLeft' });
```